packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -994,7 +994,7 @@ describe('Select', () => {
                 expect(element.value).toBe('one');
 
                 await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
-                pageObject.clickOption(2); // index 2 matches option with 'Three' text
+                pageObject.clickOption(1); // index 1 matches option with 'Three' text
```

Comment:
Should we just change these to do `clickOptionWithDisplayText()`? It'd be more readable IMO (could remove the comment).

What's the use case for `clickOption()` via index for a client? Should we just make it `internal`?

Reply:
The use case that is interesting for a client is when the display text is the same for the two options that they want to distinguish between. This can happen practically when using groups, so a client team wants to be able to click particular options that have the same name but are under different groups (for example). Using the index is an easy way to do this. We _could_ create a more direct API to click an option with a particular group parent, but at the moment I don't feel strongly compelled to do so, but would if you think that better.

Reply:
Ah that makes sense. I'm good to keep it public then and agree that we can hold off on writing `clickOptionWithGroupAndDisplayText()`.

I still think `clickOptionWithDisplayText()` would generally be preferred for readability. If you agree then we could consider:
1. noting the use cases in the docs for both functions
2. updating these tests to use the preferred method (I didn't check if these use grouping)
3. ensure we still have good test coverage for `clickOption()` if we use it less in this file